py/sentry_streams/user_functions/function_template.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+from typing import Any, Generic, Optional, TypeVar
+
+InputType = TypeVar("InputType")
+OutputType = TypeVar("OutputType")
+IntermediateType = TypeVar("IntermediateType")
+
+
+class AggregationBackend(ABC, Generic[OutputType]):
+    """
+    A storage backend that is meant to store windowed aggregates. Configurable
+    to the type of storage.
+    """
+
+    @abstractmethod
+    def flush_aggregate(self, aggregate: OutputType) -> None:
```

Comment:
Is this actually supposed to be a class ? In other word, is it supposed to handle state ? Can it be a function?

Reply:
It's been updated a bit. LMK what you think